module/condition/src/Infrastructure/Persistence/Query/DbalConditionSetQuery.php

```diff
@@ -51,15 +51,14 @@ public function getDataSet(Language $language): DbalDataSet
     /**
      * @return array
      */
-    public function findNumericConditionRelations(AttributeId $attributeId): array
+    public function findAttributeIdConditionRelations(AttributeId $attributeId): array
     {
         $qb = $this->connection->createQueryBuilder();
 
         $records = $qb->select('id')
             ->from(self::TABLE)
-            ->where($qb->expr()->eq('conditions->0->>\'type\'', ':type'))
-            ->where($qb->expr()->eq('conditions->0->>\'attribute\'', ':attribute_id'))
-            ->setParameter(':type', 'NUMERIC_ATTRIBUTE_VALUE_CONDITION')
+            ->from('jsonb_array_elements(conditions) AS condition')
```

Comment:
```
->from(self::TABLE)
->from('jsonb_array_elements(conditions) AS condition')
```
FROM table_a, table_b

In my opionion is missing a place condition `where table_a.row = table_b.row`

Reply:
It's unpacking the rows from the JSON array value by design - and unfortunately, I do not know a better way to do it in JSON clean approach. There's not matching of the rows as it's inbuilt in the mechanism itself so the way it is written I think is correct.

Alternatively `jsonb_to_recordset` could be tested against performance or maybe even text search with property name...

If this approach with unpacking is the one to go I'd suggest using DISTINCT to cleanup the result before returning it to PHP

Reply:
Ok. Query in JSON is are relatively new for me.